### ISSUE TITLE
Clarity dropdown does not convey an appropriate name, role, or state information.

### ISSUE THREAD

**Issue by bashibozuk** [vmware/clarity#3983]
## Summary

When focused the clarity dropdown is announced by the screen readers as <it's label>. button and not as dropdown menu. Also the dropdown and its items should provide appropriate aria attributes to indicate its role and state for screen readers.

* What is the change?
Implement markup for Clarity Dropdown as per W3 recommendations https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html 
* Why should it go in Clarity?
To provide accessibility for the clr dropdown
* Does this change impact existing behaviors? If so how?
No
* If this change introduces a new behavior, is this behavior accessible?
Yes

## Use case

As a screen reader user I want the screen reader to announse the clarity dropdown as a dropdown combo box for me.

## Examples

https://getbootstrap.com/docs/4.3/components/dropdowns/

## Workarounds

No

**Comment by gnomeontherun** [comment]
@bashibozuk Thanks for all of the accessibility reports like this one, could you please provide more detail and a reproduction of the issue so we know that we are talking about the same things? We can guess but it doesn't help us understand and test the issue the same way you are seeing things, which leads to longer triage times and misunderstandings.

**Comment by bashibozuk** [comment]
The aria attributes are not applied to the markup because we are using Clarity 1.1.4 which does not have these features implemented
